src/Agent.Worker/TaskRunner.cs

```diff
@@ -157,11 +155,11 @@ public async Task RunAsync()
                         Dictionary<string, VariableValue> variableCopy = new Dictionary<string, VariableValue>(StringComparer.OrdinalIgnoreCase);
                         foreach (var publicVar in ExecutionContext.Variables.Public)
                         {
-                            variableCopy[publicVar.Key] = new VariableValue(stepTarget.TranslateToHostPath(publicVar.Value));
+                            variableCopy[publicVar.Name] = new VariableValue(stepTarget.TranslateToHostPath(publicVar.Value));
```

Comment:
Do we have other instanances of this?  How was this change tested (unfortuantly, likely NOT covered by PR checks currently)

Reply:
Ah, this is caused due to changing KeyPair -> Variable class?

Reply:
Correct, changed from KeyValuePair to Variable, so Key -> Name.

Tested locally on Windows and Linux by registering a self-hosted agent and running some test pipelines against them.